### PR TITLE
findnewest: init at 0.3

### PR DESCRIPTION
**Pull request by bhipple**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg build findnewest

File: pkgs/development/tools/misc/findnewest/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, autoreconfHook }:
+
+stdenv.mkDerivation rec {
+  name = "findnewest-${version}";
```
Comment (ryantm) [inline pkgs/development/tools/misc/findnewest/default.nix]:
```suggestion
  pname = "findnewest";
```

File: pkgs/development/tools/misc/findnewest/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.3";
+
+  src = fetchFromGitHub {
+    owner = "0-wiz-0";
+    repo = "findnewest";
```
Comment (ryantm) [inline pkgs/development/tools/misc/findnewest/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/tools/misc/findnewest/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    rev = name;
+    sha256 = "1x1cbn2b27h5r0ah5xc06fkalfdci2ngrgd4wibxjw0h88h0nvgq";
+  };
+
+  buildInputs = [ autoreconfHook ];
```
Comment (ryantm) [inline pkgs/development/tools/misc/findnewest/default.nix]:
```suggestion
  nativeBuildInputs = [ autoreconfHook ];
```
Probably `autoreconfHook` is only needed at compile time.

**Review by ryantm — CHANGES_REQUESTED** [review]

**Comment by bhipple** [comment]
That's much nicer, thank you Ryan!  Should I rebase and squash these or do you have a 1-click way to do this on the PR merge button?

**Comment by ryantm** [comment]
I can use the squash and merge feature on GitHub.

File: pkgs/development/tools/misc/findnewest/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, autoreconfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "findnewest";
+  version = "0.3";
+
+  src = fetchFromGitHub {
+    owner = "0-wiz-0";
+    repo = pname;
+    rev = name;
```
Comment (ryantm) [inline pkgs/development/tools/misc/findnewest/default.nix]: Oops, I didn't realize the rev was the `name`, which is now no longer available in the attrset passed to mkDerivation. So I guess:

```suggestion
    rev = "${pname}-${version}";
```

**Review by ryantm — APPROVED** [review]

**Pull request merged**